src/Factory/OAuth2ServerFactory.php

```diff
@@ -88,6 +88,12 @@ private static function createStorageFromAdapter($adapter, array $config, Servic
             case 'mongo':
                 return self::createMongoAdapter($config, $services);
             default:
+                if ($services->has($adapter)) {
```

Comment:
I think I saw your code from a previous comment. From memory, what is done in zf-oauth2-doctrine is a complete rewrite. This change is just to cover a new backend... not the best way, you right, but it's sufficient I think. If I'm wrong, can you provide more information why you go back to your idea previously?